### ISSUE TITLE
[Question] What's the range of the example cave generation nodes tree?

### ISSUE THREAD

**Issue by Filocava99** [Auburn/FastNoise2#126]
Hi, I was trying out your tree example for cave generations (base64: EwCamZk+GgABEQACAAAAAADgQBAAAACIQR8AFgABAAAACwADAAAAAgAAAAMAAAAEAAAAAAAAAD8BFAD//wAAAAAAAD8AAAAAPwAAAAA/AAAAAD8BFwAAAIC/AACAPz0KF0BSuB5AEwAAAKBABgAAj8J1PACamZk+AAAAAAAA4XoUPw==) but I'm a bit confused on the values range and when it is supposed to be hollow. Is it solid on the min/max values or when it's very close to 0?

**Comment by Auburn** [comment]
In the NoiseTool the default setting is any value >= 0 is solid

You can see the generated min/max values and change the isosurface value in the debug window

**Comment by Filocava99** [comment]
Mmm that's weird cause I am getting a very weird result. The max value that I get from the NoiseTool is a 0.02, almost all the values are negative.
For the sampling I am using fnGenSingle3D and iterating each block of a cube (16x16x256) , it's not efficient but I'm just doing some tests.
Is there any detailed documentation for the parameters of the Gen* functions (grid, single, etc)?

**Comment by Auburn** [comment]
Sorry I got that the wrong way round, value >= 0 is air. 

Max is only 0.02 to keep the cave sizes smaller, you can change this by adjusting the Subtract node.

**Comment by Filocava99** [comment]
Hi, thanks for replying so quickly.
I tried that but the result it's the inverse of what you see in the picture (the holes are now filled and viceversa).
At this point I am wondering if I am using the generator functions correctly.

Given a cube X * Y * Z of size 16*256*16, running the following code
```
for(int x = 0; x < 16; x++){
  for(int y = 0; y < 256; y++) {
    for(int z = 0; z < 16'; z++) {
      bool isCave = fnGenSingle3D(caveNoise, x, y, z, seed) >= 0f
      ... store result ...
    }
  }
}
```
it should give me a coherent list of coordinates that form up a cave.

I also tried a simple terrain generation tree and instead of sampling each 2D coordinate I tried using GenUniformGrid2D; I tried generating two chunks, which means I run the following code:
```
int chunkX = 0, chunkZ = 0;
float[] chunk1 = fnGenUniformGrid3D(chunkNoise, arrayPointer, chunkX*16, 0, chunkZ*16,16, 256, 16, 0.1f, seed, null)
int chunkX = 1, chunkZ = 0;
float[] chunk2 = fnGenUniformGrid3D(chunkNoise, arrayPointer, chunkX*16, 0, chunkZ*16,16, 256, 16, 0.1f, seed, null)
```
In that case the surface of the terrain is coherent (meaning it doesn't make any weird jump) only inside the same chunk, making highly noticeble the connection between the two chunks.

I also wanted to describe the issue with the terrain generation cause everything is problably related to how I am using the APIs given I couldn't find any documentation.
